File: Answer/3612.用特殊操作处理字符串-i.java
/*
 * @lc app=leetcode.cn id=3612 lang=java
 *
 * [3612] 用特殊操作处理字符串 I
 */

// @lc code=start
class Solution {

    public String processStr(String s) {
        StringBuilder sb = new StringBuilder();
        for (char ch : s.toCharArray()) {
            if (ch == '*' && sb.length() > 0) {
                sb.deleteCharAt(sb.length() - 1);
            } else if (ch == '#') {
                sb.append(sb);
            } else if (ch == '%') {
                sb.reverse();
            } else if (ch >= 'a' && ch <= 'z') {
                sb.append(ch);
            }
        }
        return sb.toString();
    }
}
// @lc code=end

File: Answer/3612.用特殊操作处理字符串-i.cs
using System.Text;

/*
 * @lc app=leetcode.cn id=3612 lang=csharp
 *
 * [3612] 用特殊操作处理字符串 I
 */

// @lc code=start
public class Solution
{
    public string ProcessStr(string s)
    {
        var sb = new StringBuilder();
        foreach (var ch in s)
        {
            if (ch == '*')
            {
                if (sb.Length > 0)
                {
                    sb.Length--;
                }
            }
            else if (ch == '#')
            {
                sb.Append(sb.ToString());
            }
            else if (ch == '%')
            {
                for (int l = 0, r = sb.Length - 1; l < r; l++, r--)
                {
                    char tmp = sb[l];
                    sb[l] = sb[r];
                    sb[r] = tmp;
                }
            }
            else
            {
                sb.Append(ch);
            }
        }
        return sb.ToString();
    }
}
// @lc code=end
